### PR TITLE
add the `.verify` property correctly

### PR DESCRIPTION
**Pull request by adehad**
Addressed the concerns raised by #339 

http://docs.python-requests.org/en/master/user/advanced/#ssl-cert-verification

`pip install git+https://github.com/pycontribs/jira.git@bugfix/correctly_add_ssl_cert`

### CODE REVIEW

File: jira/client.py [hunk 1]
```diff
@@ -3403,6 +3395,13 @@ def _add_client_cert_to_session(self):
         client_cert: Tuple[str, str] = self._options["client_cert"]  # to help mypy
         self._session.cert = client_cert
 
+    def _add_ssl_cert_to_session(self):
+        """
+        Adds the client certificate to the request if configured through the constructor.
```
Comment (kchason) [inline jira/client.py:3400]: Is this adding the client certificate, or the certificate authority?

Comment (adehad) [reply jira/client.py]: Thanks, this was a copy paste job I failed to update. I've now updated both the name of the function and improved the docstring for clarity.

File: jira/client.py [hunk 2]
```diff
@@ -480,7 +483,6 @@ def __init__(
             self._create_oauth_session(oauth, timeout)
         elif basic_auth:
             self._create_http_basic_session(*basic_auth, timeout=timeout)
-            self._session.headers.update(self._options["headers"])
```
Comment (studioj) [inline jira/client.py:483]: i see this is removed but not added somewhere else ....

**Review by studioj — CHANGES_REQUESTED** [review]
I would maybe look for adding a test for this... this has the potential to break a lot :-) and probably will sneak through CI without a proper test :-S. I wouldn't know how to test it though :-D

File: jira/client.py (hunk 2 above)
Comment (adehad) [reply jira/client.py:483]: I removed this as it is already called in L502 (current main) regardless of the auth session type:

https://github.com/pycontribs/jira/blob/db56d5c98dd5a792e6a347c3ff87f7f8cc752599/jira/client.py#L499-L502

**Comment by adehad** [comment]
>I would maybe look for adding a test for this... this has the potential to break a lot :-) and probably will sneak through CI without a proper test :-S. I wouldn't know how to test it though :-D

Yeah I have no idea how to test this, really the `ResilientSession` class should accept these arguments as parameters and set it during the `__init__`.

**Review by studioj — APPROVED** [review]

**Comment by studioj** [comment]
> > I would maybe look for adding a test for this... this has the potential to break a lot :-) and probably will sneak through CI without a proper test :-S. I wouldn't know how to test it though :-D
> 
> Yeah I have no idea how to test this, really the `ResilientSession` class should accept these arguments as parameters and set it during the `__init__`.

:-) see it as a challenge :D you have my approval...

**Comment by adehad** [comment]
>:-) see it as a challenge :D

😆

**Pull request merged**